modules/hardware/common/usb/vhotplug.nix

```diff
@@ -17,137 +17,208 @@ let
-        }
-      ];
-    }
-    {
-      description = "Network Devices for NetVM";
-      targetVm = "net-vm";
-      allow = [
-        {
-          interfaceClass = 2;
-          interfaceSubclass = 6;
-          description = "Communications - Ethernet Networking";
-        }
-        {
-          vendorId = "0b95";
-          productId = "1790";
-          description = "ASIX Elec. Corp. AX88179 UE306 Ethernet Adapter";
-        }
-      ];
-    }
-  ]
-  ++
-    optionals
-      (
-        config.ghaf.virtualization.microvm.appvm.enable
-        && config.ghaf.virtualization.microvm.appvm.vms.chrome.enable
-      )
-      [
-        {
-          description = "External Webcams for ChromeVM and BusinessVM";
-          allowedVms = [
-            "chrome-vm"
-            "business-vm"
-          ];
-          allow = [
-            {
-              interfaceClass = 14;
-              description = "Video (USB Webcams)";
-            }
-          ];
-          deny = [
-            {
-              # Ignore Lenovo X1 camera since it is attached to the business-vm
-              # Finland SKU
-              vendorId = "04f2";
-              productId = "b751";
-              description = "Lenovo X1 Integrated Camera";
-            }
-            {
-              # Ignore Lenovo X1 camera since it is attached to the business-vm
-              # Uae 1st SKU
-              vendorId = "5986";
-              productId = "2145";
-              description = "Lenovo X1 Integrated Camera";
-            }
-            {
-              # Ignore Lenovo X1 camera since it is attached to the business-vm
-              # UAE #2 SKU
-              vendorId = "30c9";
-              productId = "0052";
-              description = "Lenovo X1 Integrated Camera";
-            }
-            {
-              # Ignore Lenovo X1 gen 12 camera since it is attached to the business-vm
-              # Finland SKU
-              vendorId = "30c9";
-              productId = "005f";
-              description = "Lenovo X1 Integrated Camera";
-            }
-            {
-              # Ignore System76 darp11-b camera since it is attached to the business-vm
-              # Finland SKU
-              vendorId = "04f2";
-              productId = "b729";
-              description = "System76 darp11-b Integrated Camera";
-            }
+  defaultUsbRules =
+    optionals config.ghaf.virtualization.microvm.guivm.enable [
+      {
+        description = "Devices for GUIVM";
+        targetVm = "gui-vm";
+        allow = [
+          {
+            interfaceClass = 3;
+            interfaceProtocol = 1;
+            description = "HID Keyboard";
+          }
+          {
+            interfaceClass = 3;
+            interfaceProtocol = 2;
+            description = "HID Mouse";
+          }
+          {
+            interfaceClass = 11;
+            description = "Chip/SmartCard (e.g. YubiKey)";
+          }
```

Comment:
does this allow multiplexing? e.g. in the case of the drone fmo laptop there is a need to pass the yubikey also to the docker-vm. so there should be a way to achieve this, like the recent usb assignment gui from Ganga